MetabolicAtlas/MetabolicAtlas
# feat: GotEnzymes database handling and deployment

**Related issue(s) and PR(s)**  
This PR closes #1023.



**Type of change**  

 
**List of changes made**  

- Update database schema including indexes.
- Use more performant lookup for semicolon separate fields (`ec_number` field from the `enzymes` table).
- Add remote docker compose configuration for the GotEnzymes database (`pg` container).
- Remove `km` from the enzymes table on the frontend
- Fix a bug where enzymes table are requested twice on initial page load
- Add section about GotEnzymes in the README.

**Testing**  

- Relative urls that can be reused both for production and local testing
- Instructions on how to test

1. Reconstruct the database: `ma-exec pg psql -f scripts/init.sql -U postgres`. This should take a while, it was around ~10 minutes for me. The latest input data can be found [here](https://nbisweden.slack.com/archives/C01D13EKUV7/p1655291764755059). **Please note** that the `enzymes` input data may need to be reformatted (removing last tab character of each line) before they can be imported. To do this, run the following command in the `enzymes` input data directory (`pg/input_data/enzymes/`), `sed -i 's/\t$//' *.txt` (On macOS this may need to be `gsed` depending on what's installed.).
2. Visit an entity's details page. For example:
  - `/enzymedb/compound/C00002`
  - `/enzymedb/ec/2.5.1.19`
  - `/enzymedb/reaction/R00006`
3. Verify that the basic info and cross references (not available for ec) tables are displayed.
4. Verify that the enzymes table is displayed. Please try out the filters (must be exact match), table sorting and pagination and verify that they work as expected.

**Further comments**  

- There is [some comments](https://github.com/MetabolicAtlas/MetabolicAtlas/issues/1023#issuecomment-1158795112) on the issue page that includes some more notes of research and findings when working on this issue.
- We were discussing the possibility of using a hash or flag to indicate if the deploy should include the database import. It seems that the deployment setup proposed in this PR has minimal impact on the deployment time so I don't think it's currently necessary to add this. More thorough explanation is provided in the comments linked above.

**Definition of Done checklist**
